WangscapeGui/mainwindow.cpp

```diff
@@ -0,0 +1,202 @@
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+
+#include <memory>
+
+#include<boost/filesystem.hpp>
+
+#include <QMessageBox>
+#include <QFileDialog>
+
+#include "logging/Logger.h"
+#include "logging/Logging.h"
+#include "logging/ConsoleAppender.h"
+#include "logging/FileAppender.h"
+#include "tilegen/TileGenerator.h"
+#include "tilegen/partition/TilePartitionerNoise.h"
+
+
+MainWindow::MainWindow(QWidget *parent_) :
+    QMainWindow(parent_),
+    mUi(new Ui::MainWindow),
+    mOptionsEditor(new OptionsEditor(this))
+{
+    mScene = new QGraphicsScene(this);
+
+    mUi->setupUi(this);
+    mUi->progressBar->setRange(0, 100);
+    mUi->progressBar->setValue(0);
+
+    openOptionsEditor();
+
+    connect(mUi->generateButton, SIGNAL(pressed()), this, SLOT(generate()));
+    connect(mUi->comboBox, SIGNAL(activated(const QString&)), this, SLOT(displayTilesetPreview(const QString&)));
+    connect(mUi->actionLoadOptions, SIGNAL(triggered()), this, SLOT(loadOptionsFromFile()));
+    connect(mUi->actionSaveOutput, SIGNAL(triggered()), this, SLOT(saveOutput()));
+    connect(mUi->actionOpenOptionsEditor, SIGNAL(triggered()), this, SLOT(openOptionsEditor()));
+}
+
+MainWindow::~MainWindow()
+{
+    delete mUi;
+}
+
+void MainWindow::generate()
+{
+    if (mTilesetGenerator == nullptr)
+    {
+        QMessageBox error_message;
+        error_message.critical(this, "Error during tileset generation", QString("You have to select options file before clicking 'Generate...' button"));
+        return;
+    }
+
+    resetTilesetGenerator();
+
+    mUi->comboBox->clear();
+    mPreviewImages.clear();
+
+    mTilesetGenerator->generate([this](const sf::Texture& output, std::string filename)
+    {
+        const sf::Image source_image = output.copyToImage();
+
+        const auto basename = filename.substr(filename.rfind('/') + 1);
```

Comment:
Can we use `boost::filesystem::path` here? This will break if `\` is used as a path separator.